json-schema-core/src/main/java/io/ballerina/jsonschema/core/Generator.java

```diff
@@ -228,6 +333,258 @@ String convert(Object schemaObject, String name) throws Exception {
+
+        String fullDeclaration = AT + ANNOTATION_MODULE + COLON + ALL_OF + NEW_LINE +
+                String.format(TYPE_FORMAT, name, mainType + PIPE + subTypesName);
+        ModuleMemberDeclarationNode moduleNodeMain = NodeParser.parseModuleMemberDeclaration(fullDeclaration);
+        nodes.put(name, moduleNodeMain);
+        return name;
+    }
+
+    private static void extractCombiningSchemas(Schema schema) {
+        // Handle if-then-else
+        List<Object> combinedSchema = new ArrayList<>(List.of());
+        int keywordCount = 0;
+
+        List<Object> ifResolved = new ArrayList<>();
+        Object ifCondition = schema.getIfKeyword();
+        Object thenCondition = schema.getThen();
+        Object elseCondition = schema.getElseKeyword();
+
+        if (ifCondition != null && (thenCondition != null || elseCondition != null)) {
+            thenCondition = thenCondition == null ? true : thenCondition;
+            elseCondition = elseCondition == null ? true : elseCondition;
+
+            List<Object> allOfList1 = new ArrayList<>();
+            allOfList1.add(ifCondition);
+            allOfList1.add(thenCondition);
+            Schema allOf1 = new Schema();
+            allOf1.setAllOf(allOfList1);
+            ifResolved.add(allOf1);
+
+            List<Object> allOfList2 = new ArrayList<>();
+            Schema notSchema = new Schema();
+            notSchema.setNot(deepCopy(ifCondition));
+            allOfList2.add(notSchema);
+            allOfList2.add(elseCondition);
+            Schema allOf2 = new Schema();
+            allOf2.setAllOf(allOfList2);
+            ifResolved.add(allOf2);
+
+            if (schema.getOneOf() != null) {
+                Schema ifOneOf = new Schema();
+                ifOneOf.setOneOf(ifResolved);
+                combinedSchema.add(ifOneOf);
+                keywordCount++;
+            } else {
+                schema.setOneOf(ifResolved);
+            }
+
+            schema.setIfKeyword(null);
+            schema.setThen(null);
+            schema.setElseKeyword(null);
+        }
+
+        // Handle AllOf, OneOf, AnyOf
+        if (schema.getAnyOf() != null) {
+            keywordCount++;
+            Schema anyOfSchema = new Schema();
+            anyOfSchema.setAnyOf(schema.getAnyOf());
+            combinedSchema.add(anyOfSchema);
+        }
+        if (schema.getOneOf() != null) {
+            keywordCount++;
+            Schema oneOfSchema = new Schema();
+            oneOfSchema.setOneOf(schema.getOneOf());
+            combinedSchema.add(oneOfSchema);
+        }
+        if (schema.getAllOf() != null) {
+            keywordCount++;
+            Schema allOfSchema = new Schema();
+            allOfSchema.setAllOf(schema.getAllOf());
+            combinedSchema.add(allOfSchema);
+        }
+
+        if (keywordCount > 1) {
+            schema.setAllOf(combinedSchema);
+            schema.setAnyOf(null);
+            schema.setOneOf(null);
+        }
```

Comment:
Shall we add a test case for nested subschemas
* all of inside a not
* any of inside a all of and one of